demo/components/opt-in-flyout/index.html

```diff
@@ -0,0 +1,94 @@
+			import '@brightspace-ui/core/components/demo/demo-page.js';
+			import '../../../src/components/opt-in-flyout/opt-in-flyout.js';
+			import '../../../src/components/opt-in-flyout/opt-out-flyout.js';
+			import '../../../src/components/opt-in-flyout/opt-out-reason.js';
+		</script>
+		<title>opt-in-flyout demo</title>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta charset="UTF-8">
+		<style>
+			.flyout-wrapper {
+				height: 450px;
+				position: relative;
+				width: 100%;
+			}
+		</style>
+	</head>
+	<body>
+		<d2l-demo-page page-title="Opt-in Flyout">
+
+			<d2l-demo-snippet no-padding>
+				<div class="flyout-wrapper">
+					<d2l-labs-opt-in-flyout
+						open
+						title="Flyout Demo Opt-in"
+						short-description="This is a <b>short</b> description" long-description="This is a <b>long</b> description"
+						tab-position="right"
+						tutorial-link="https://www.example.com#tutorial"
+						help-docs-link="https://www.example.com#documentation">
+					</d2l-labs-opt-in-flyout>
+				</div>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<div class="flyout-wrapper">
+					<d2l-labs-opt-out-flyout
+						open
+						title="Flyout Demo Opt-out"
+						short-description="This is a <b>short</b> description"
+						long-description="This is a <b>long</b> description"
+						tab-position="right"
+						tutorial-link="https://www.example.com#tutorial"
+						help-docs-link="https://www.example.com#documentation">
+					</d2l-labs-opt-out-flyout>
+				</div>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<div class="flyout-wrapper">
+					<d2l-labs-opt-out-flyout
+						open
+						title="Flyout Demo Opt-out"
+						short-description="This is a <b>short</b> description"
+						long-description="This is a <b>long</b> description"
+						tab-position="right"
+						tutorial-link="https://www.example.com#tutorial"
+						help-docs-link="https://www.example.com#documentation">
+						<d2l-labs-opt-out-reason key="test-1" text="Test Option 1"></d2l-labs-opt-out-reason>
+						<d2l-labs-opt-out-reason key="test-2" text="Test Option 2"></d2l-labs-opt-out-reason>
+					</d2l-labs-opt-out-flyout>
+				</div>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<div class="flyout-wrapper">
+					<d2l-labs-opt-out-flyout
+						hide-reason
+						hide-feedback
+						open
+						title="Flyout Demo Opt-out - dialog disabled"
+						short-description="This is a <b>short</b> description"
+						long-description="This is a <b>long</b> description"
+						tab-position="right"
+						tutorial-link="https://www.example.com#tutorial"
+						help-docs-link="https://www.example.com#documentation">
+					</d2l-labs-opt-out-flyout>
+				</div>
+			</d2l-demo-snippet>
+
+			<script type="module">
+				[...document.querySelectorAll('d2l-labs-opt-in-flyout, d2l-labs-opt-out-flyout')].forEach(elem => {
+					['flyout-opened', 'flyout-closed', 'opt-in', 'opt-out'].forEach(eventName => {
```

Comment:
If the demo snippet HTML isn't wrapped in a `<template>` then the formatted code that is displayed will contain any properties that are reflected as attributes which might not actually be part of the public API. This can be somewhat misleading to anyone looking at the demo code snippet so wrapping in a `<template>` is generally preferred. 

However in doing so, the rendering of the snippet elements becomes delayed due to the demo-snippet stamping and Lit's async rendering. This can result in JavaScript errors trying to query for the elements immediately. Three solutions I've used for that - 1) wrap the demo event code in a `setTimeout(...)` so that it's similarly delayed, or 2) wire up to `document.body` instead if the events bubble, which simplifies the script code slightly more; or 3) move the `<script>` containing the event wire-up into the `<template>` if we want to show that to the developer, though that means repeating it for each snippet and assigning ids so it can identify the correct element to wire up to.

So long story short, if we want this to be an example for other labs to copy, and we want them to use the `<template>` approach, then this script code would need to be tweaked slightly.

Reply:
Interesting. I'll update the create template with this as well. I've updated it to wrap in a `<template>` and this code is actually still working... so... win?

Reply:
Cool. Yeah so win!

I did a quick test with reflecting an internal property and it wasn't showing up in the formatted code, so maybe that got resolved as a result of some other changes to `d2l-demo-snippet` over the years. However, I did confirm that the constructor runs twice when `template` isn't used as a result of the way that `d2l-demo-snippet` grabs the content, clone and appends it internally, formats, etc., whereas the constructor only runs once when a `template` is used. So using `template` still seems preferred.

Reply:
Gotcha. Now that vdiff is using `@web/test-runner`, it'll be interesting to see if we can update the demo snippet such that you could pass it a fixture that's defined elsewhere as an HTML literal, which would allow us to reuse these between demos and tests.

Reply:
One potential negative effect of always using `template` and having the delayed rendering - for the `SkeletonMixin` stuff, it "fixed" the crazy import order problem we were seeing, but doesn't _really_ fix it since actually consumers aren't using a `template`.  Something we might just need to keep in mind, in theory tests would (hopefully) highlight the issue